core/dbt/task/show.py

```diff
@@ -23,14 +24,21 @@ def execute(self, compiled_node, manifest):
         # Allow passing in -1 (or any negative number) to get all rows
         limit = None if self.config.args.limit < 0 else self.config.args.limit
 
-        if "sql_header" in compiled_node.unrendered_config:
-            compiled_node.compiled_code = (
-                compiled_node.unrendered_config["sql_header"] + compiled_node.compiled_code
-            )
-
+        model_context = generate_runtime_model_context(compiled_node, self.config, manifest)
+        compiled_node.compiled_code = self.adapter.execute_macro(
+            macro_name="get_show_sql",
+            manifest=manifest,
+            context_override=model_context,
+            kwargs={
+                "compiled_code": model_context["compiled_code"],
+                "sql_header": model_context["config"].get("sql_header"),
+                "limit": limit,
+            },
+        )
         adapter_response, execute_result = self.adapter.execute(
-            compiled_node.compiled_code, fetch=True, limit=limit
```

Comment:
no longer passing limit through self.adapter.execute or making use of fetchmany, so closing https://github.com/dbt-labs/dbt-core/issues/8471 as part of this PR as well